lib/GRNOC/WebService/Method.pm

```diff
@@ -1409,7 +1409,24 @@ sub _return_error{
   my %error;
   $error{"error"}       = 1;
   $error{'error_text'}  = $self->get_error();
-  $error{'results'}     = undef;
+  $error{'results'}   = undef;
+
+  my $allow_credentials='false';
+  my $allowed_origin= "http://grnoc.iu.edu"; #this fails most of the time
+  my $regexp=$self->{'xdr_regexp'};
+  if ( $cgi->http('ORIGIN') =~ /$regexp/) {
+
+    $allowed_origin=$cgi->http('ORIGIN');
+    if (defined $ENV{'HTTPS'}) {
+      $allow_credentials='true';
+    }
+  }
+  print $cgi->header(-Access_Control_Allow_Origin => $allowed_origin,
```

Comment:
Calling `$cgi->header` twice results in two sets of http headers being written. This results in the second set of http headers being included in the http body. Combine both `$cgi->header` calls.